apps/ejabberd/src/mod_amp.erl

```diff
@@ -45,21 +45,39 @@ stop(Host) ->
 
 %% Business API
 
--spec check_packet(exml:element(), amp_event()) -> exml:element() | drop.
+-spec check_packet(mongoose_acc:t() | exml:element(), amp_event()) ->
+    mongoose_acc:t() | exml:element() | drop.
 check_packet(Packet = #xmlel{attrs = Attrs}, Event) ->
+    % it is called this way only from ejabberd_c2s:send_and_maybe_buffer_stanza/3, line 1666
+    % maybe Paweł Chrząszcz knows why and can advise what to do about it
```

Comment:
Yes, I will, but now I'm in the middle of such a mess that it doesn't make much sense to talk to him right now. Those comments are rather as a reminder, what we should have him look at.